module1.js

```diff
@@ -152,11 +156,12 @@ function question1(name1) {
   @returns { totalPrice } a string containing the currency sign too
 */
 
-// function question9( retailPrice ) {
-//   // TODO
-  
-//   return /* ??? */;
-// }
+function question9(retailPrice, taxAmount) {
+  let rpStringToNumber = parseFloat(retailPrice.replace(/[^0-9-.]/g, ''))
+  let res = (taxAmount + 1) * rpStringToNumber;
+  return `£${res}`
+}
+//how to ceovert it to es6???
 
```

Comment:
A good solution using the regex to remove the currency symbol. Probably a much better option than using something like `str.split('£');`, which would have given you an array of `["", '10"]` as a result.

For the ES6, you're probably already using as of it as you can here - there isn't always an alternative to ES5 syntax.

You could also present the code as follows which may help with readability
```
function question9(retailPrice, taxAmount) {
    const nonNum = /[^0-9-.]/g; // regex to remove all non-numeric chars
    const rpStringToNumber = parseFloat(retailPrice.replace(nonNum, ''));
    const res = (taxAmount + 1) * rpStringToNumber;
    return `£${res}`;
}
```
You can use `const` in this function instead of `let`, as the variables declared inside the function are never re-assigned. Each time the function is called, they are re-defined afresh, so you won't get an error from them.

Reply:
Thanks for your all the feedbacks
using str.split('£') also can be kinda hard coding, using regex can be universal even for $ ,...